README.md

```diff
@@ -1 +1,167 @@
-# witnet-block-relay
\ No newline at end of file
+# witnet-ethereum-block-relay
+
+`witnet-ethereum-block-relay` is an open source implementation of the Block Relay in Witnet.
+
+ DISCLAIMER: this is a work in progress, meaning the contract could be voulnerable to attacks.
+
+## About WBI and the Block Relay
+
+The Bridge nodes interact through the Witnet Bridge Interface, a smart contract in charge of facilitating the communication between Witnet and samrt contract platmforms, such as Ethereum (click [here][wbi] for more information).
+The block headers in Witnet need to be available to the WBI, meaning the WBI should get those headers that should be stored in some contract, the Block Relay, in Witnet. In order to assure the validity of the block headers, the Block Relay needs to guarantee a consensus protocol and a Finality Gadget. The documentation about the stages of the Block Relay can be found [here][block-relay].
+
+## Connecting the Bridge and the Block Relay
+
+Due to the above, the more isolated the Block Relay and the WBI are, the more secure. Considering this, a proxy contract has been implemented, which connects to the WBI and from which the Block Relay is called. This allows to upgrade the Block Relay when needed.
+The following diagram shows the flow between the WBI and the Block Relay.
+
+<p align=center>
+<img src="./images/wbi_br.jpg" width="550">
+</p>
+<p align=center>
+<em>Fig. 1: Connecting the WBI to the Block Relay.</em>
+</p>
+
+## BlockRelay based on ABS
+
+The `ABSBlockRelay` is a work in progress contract that pretends to implement the Active Bridge Set (ABS) as the set in charge of proposing and posting blocks to the Block Relay. The ABS is updated in the WBI and is a subset of bridge nodes.
+
+A block header is later finalized when 2/3 of the ABS agree on a vote. More precisely the flow is as follows:
```

Comment:
A block header is later finalized when 2/3 of the ABS agree on a vote. More precisely, the flow is as follows: